NSURL.m

```diff
@@ -351,7 +363,41 @@ static CFTypeRef CFURLCreatePropertyForKey(CFURLRef url, CFStringRef key, CFErro
     else if (CFEqual(key, kCFURLTypeIdentifierKey))
     {
         // Key for the resource’s uniform type identifier (UTI), returned as a CFString object.
+        static dispatch_once_t pred;
+        static void* LaunchServicesHandle;
+        static CFErrorRef* LaunchServicesHandleError;
+
+        dispatch_once(&pred, ^{
+            LaunchServicesHandle = dlopen("/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/LaunchServices", RTLD_LAZY | RTLD_LOCAL);
+            if(LaunchServicesHandle == NULL)
+                dlError(LaunchServicesHandleError);
+            });
+
+        if(LaunchServicesHandle == NULL){
+            error = LaunchServicesHandleError; 
```

Comment:
If someone calls this function multiple times, they'll get the same error reference, but that's wrong because they have to `CFRelease` it, which would delete our reference on it; you want to `CFRetain` it here before passing it out.

Also, you want to assign to the pointed error (if present) instead of the error pointer itself:
```suggestion
            if (error != NULL)
                *error = CFRetain(LaunchServicesHandleError);
```